### PR TITLE
feat: add debug parameter for core

### PR DESCRIPTION
**Pull request by shuplenkov**
## What was done?

New `debug` parameter added to core

```mn config:set --config=local_seed core.debug true```

## Breaking Changes

No

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: configs/schema/configJsonSchema.js [hunk 1]
```diff
@@ -195,6 +195,9 @@ module.exports = {
           type: ['string', 'null'],
           minLength: 1,
         },
+        debug: {
```
Comment (shumkov) [inline configs/schema/configJsonSchema.js:198]: I would keep it boolean only and set the default value in the base config. Also, please update the migration.

File: configs/schema/configJsonSchema.js [hunk 2]
```diff
@@ -195,6 +195,9 @@ module.exports = {
...
+          type: ['boolean'],
+        },
       },
       required: ['docker', 'p2p', 'rpc', 'spork', 'masternode', 'miner', 'devnetName'],
```
Comment (shumkov) [inline configs/schema/configJsonSchema.js:199]: `debug` must be here as well

**Review by shumkov — CHANGES_REQUESTED** [review]

File: configs/migrations.js [hunk 3]
```diff
@@ -140,6 +140,11 @@ module.exports = {
         config.core.docker.image = systemConfigs.base.core.docker.image;
         config.platform.dapi.api.docker.image = systemConfigs.base.platform.dapi.api.docker.image;
         config.platform.drive.abci.docker.image = systemConfigs.base.platform.drive.abci.docker.image;
+
+        // Core debug
+        if (typeof config.code.debug === 'undefined') {
```
Comment (antouhou) [inline configs/migrations.js:145]: core

Comment (shumkov) [inline configs/migrations.js:145]: `config.core.debug`

**Review by shumkov — APPROVED** [review]
LGTM! 👍

File: configs/migrations.js [hunk 4]
```diff
@@ -140,6 +140,11 @@ module.exports = {
         config.core.docker.image = systemConfigs.base.core.docker.image;
         config.platform.dapi.api.docker.image = systemConfigs.base.platform.dapi.api.docker.image;
         config.platform.drive.abci.docker.image = systemConfigs.base.platform.drive.abci.docker.image;
+
+        // Core debug
+        if (typeof config.core.debug === 'undefined') {
```
Comment (shumkov) [inline configs/migrations.js:145]: Since we merge it to 0.20 you need to add migration for the new version.

Comment (shumkov) [inline configs/migrations.js:145]: Since we merge it to 0.20 you need to add migration for the new version.

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — DISMISSED** [review]
👍

**Review by shumkov — APPROVED** [review]
👍

**Pull request merged**